### ISSUE TITLE
setup-gcloud should accept both base64-encoded keys and non-encoded keys

### ISSUE THREAD

**Issue by jbaldassari** [GoogleCloudPlatform/github-actions#129]
### TL;DR

The  `setup-gcloud` action requires base64-encoded keys as its `service_account_key` parameter, while most other actions (e.g. `get-secretmanager-secrets`) will accept both base64-encoded and non-encoded JSON keys with the documentation specifically asking for non-encoded keys.

### Design

I can think of a couple possible solutions to this issue.  One would be for `setup-gcloud` to detect whether the key is base64-encoded or not as some of the other actions do.  Another would be to provide two separate input parameters, one that accepts base64-encoded values and one that expects the value to not be encoded.

**Resources**

https://github.com/GoogleCloudPlatform/github-actions/blob/master/setup-gcloud/README.md

https://github.com/GoogleCloudPlatform/github-actions/blob/master/get-secretmanager-secrets/README.md

https://github.com/GoogleCloudPlatform/github-actions/blob/master/get-secretmanager-secrets/src/client.ts#L49-L55

**Comment by sethvargo** [comment]
It'd probably be worth centralizing our "service account key parsing" somewhere /cc @averikitsch

**Comment by averikitsch** [comment]
The documentation for setup-gcloud needs to be updated to reflect that it can use both encode and non-encode credentials. The setupGcloudSdk folder contains methods for key parsing, so we should discuss a plan to utilizing that with our all our actions.

**Comment by jbaldassari** [comment]
Hi @averikitsch.  Thanks for clarifying.  What was happening was that our workflow was taking the value of an organization secret and setting it to an environment variable so that the rest of the workflow didn't have to know which org secret was being used.  I think the issue with the non-encoded key was that the newlines and quotes were causing problems with the github `::set-env` command.  It's possible that it would have worked if we had passed the org secret directly to `setup-gcloud`.
